### PR TITLE
Fixed schema refs for ml.yaml

### PR DESCRIPTION
**Pull request by nhtruong**
The body schemas for some bodies in the `ml` namespace were incorrectly referenced to the content-type instead of their respective `schema` property.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — APPROVED** [review]
Should the checker have caught this as additional fields or something?

**Review by dblock — CHANGES_REQUESTED** [review]
Looks like ML tests started failing.

**Comment by nhtruong** [comment]
> Should the checker have caught this as additional fields or something?

We should create JSON-schemas for the namespace and category files to catch this among others, especially on catching typos and enforcing rules for extensions.

**Comment by dblock** [comment]
> > Should the checker have caught this as additional fields or something?
> 
> We should create JSON-schemas for the namespace and category files to catch this among others, especially on catching typos and enforcing rules for extensions.

Yes, you want to take it? https://github.com/opensearch-project/opensearch-api-specification/issues/492

The tests here are failing though, so you'll need to add a fix.

**Comment by nhtruong** [comment]
> The tests here are failing though, so you'll need to add a fix.

I saw that. I'm on it.
It also reveals a potential bug in the test framework: When the response schema is not defined in the spec (At least the framework thought so in this case because the schema was referenced in the wrong place), it skips response payload test and still pass the chapter. I think it should throw an error (spec not found) in this case just like we will fail the test if a property is missing (In this case all fields are missing 😆 yet the test passed)

**Comment by dblock** [comment]
> > The tests here are failing though, so you'll need to add a fix.
> 
> I saw that. I'm on it. It also reveals a potential bug in the test framework: When the response schema is not defined in the spec (At least the framework thought so in this case because the schema was referenced in the wrong place), it skips response payload test and still passes the chapter. I think it should throw an error (spec not found) in this case just like we will fail the test if a property is missing (In this case all fields are missing 😆 yet the test passed)

Feel free to open this one separately and I can take care of it.

**Comment by nhtruong** [comment]
@dblock I think I've found another [bug](https://github.com/opensearch-project/opensearch-api-specification/actions/runs/10357629378/job/28670062845?pr=489#step:7:66):
> Skipped because version 2.17.0 does not satisfy >= 2.11, < 3.0.

File: spec/namespaces/ml.yaml [hunk 1]
```diff
@@ -130,36 +130,52 @@ components:
...
       content:
         application/json:
-          $ref: '../schemas/ml._common.yaml#/components/schemas/ModelGroup'
+          schema:
+            $ref: '../schemas/_common.yaml#/components/schemas/WriteResponseBase'
     ml.register_model@200:
       content:
         application/json:
-          $ref: '../schemas/ml._common.yaml#/components/schemas/Task'
+          schema:
```
Comment (dblock) [inline spec/namespaces/ml.yaml:154]: Is this not a task? In tests we register a model then wait it the task to complete.

**Review by dblock — CHANGES_REQUESTED** [review]
See below.

File: spec/namespaces/ml.yaml (hunk 1 above)
Comment (nhtruong) [reply spec/namespaces/ml.yaml:154]: The response is not the entire Task object, which is different from `get_task`'s response. This only returns the task_id for you to check on it later.

**Pull request merged**